src/HttpCache/VarnishXKeyPurger.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\HttpCache;
+
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Purges Varnish XKey.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @experimental
+ */
+final class VarnishXKeyPurger implements PurgerInterface
+{
+    private const VARNISH_MAX_HEADER_LENGTH = 8000;
+
+    private $clients;
+    private $maxHeaderLength;
+
+    /**
+     * @param ClientInterface[] $clients
+     */
+    public function __construct(array $clients, int $maxHeaderLength = self::VARNISH_MAX_HEADER_LENGTH)
+    {
+        $this->clients = $clients;
+        $this->maxHeaderLength = $maxHeaderLength;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function purge(array $iris)
+    {
+        if (!$iris) {
+            return;
+        }
+
+        $irisChunks = array_chunk($iris, \count($iris));
+
+        foreach ($irisChunks as $irisChunk) {
+            // TODO: glue could be a parameter in a config file
+            $this->purgeRequest($irisChunk, ' ');
+        }
+    }
+
+    private function purgeRequest(array $iris, string $glue)
+    {
+        foreach ($this->chunkKeys($iris, $glue) as $keys) {
+            $this->purgeKeys($keys);
+        }
+    }
+
+    private function purgeKeys(string $keys): void
+    {
+        foreach ($this->clients as $client) {
+            $client->request('PURGE', '', ['headers' => ['ApiPlatform-Purge-Keys' => $keys]]);
+        }
+    }
+
+    private function chunkKeys(array $keys, string $glue): iterable
+    {
+        $concatenatedKeys = implode(' ', $keys);
+
+        // If all keys fit in the header, we can return them
+        if (\strlen($concatenatedKeys) <= $this->maxHeaderLength) {
+            yield $concatenatedKeys;
+
+            return;
+        }
+
+        $currentHeader = '';
+
+        foreach ($keys as $position => $key) {
+            // Key can be added to header
+            if (\strlen($currentHeader.($position > 0 ? $glue : '').$key) <= $this->maxHeaderLength) {
+                $currentHeader .= ($position > 0 ? $glue : '').$key;
+            } else {
+                if (\strlen($key) > $this->maxHeaderLength) {
+                    // TODO: rename this exception to better represent the error
+                    throw new \Exception(sprintf('Path : %s is too long to fit current max header length (currently set at %s). You must increase max header length.', $key, $this->maxHeaderLength));
```

Comment:
```suggestion
                    throw new \Exception(sprintf('Path: %s is too long to fit current max header length (currently set to %s). You must increase max header length.', $key, $this->maxHeaderLength));
```
This exception is weird since the header length cannot be modified by the user.